package.json

```diff
@@ -75,6 +64,17 @@
 	},
 	"devDependencies": {
 		"@sindresorhus/tsconfig": "0.7.0",
+		"@types/is-ci": "^2.0.0",
+		"@types/lodash.throttle": "^4.1.6",
```

Comment:
You cannot move all types. This type, for example, is used in the source code: https://github.com/vadimdemedes/ink/blob/293d40fb6f85478ae0b8120709d71bf530d23eb2/src/ink.tsx#L3

Reply:
That is only used in the compiled js, not in the TypeScript definitions, so it's not required by this package at runtime nor at downstream compile time :)

Reply:
Ok I think I'm confused now. What's a question I need to ask myself to determine when a `@types/*` dependency should be included in `dependencies` or `devDependencies`? My understanding was that this question is "Are types from @types/* package are re-exported from my module in any way?".

Reply:
> "Are types from @types/* package re-exported from my module in any way?".

That is the correct question to ask. 

I'm not sure why @sindresorhus thinks this type is required in `dependencies`, since it fails the above question.

Reply:
You need to keep at least `react`, `node`, and `yoga-layout` (the latter seems to ship its own types)

![image](https://user-images.githubusercontent.com/1404810/86542295-d90d4c80-bf14-11ea-9adf-69b721c7b981.png)

Reply:
Yeah, sorry. I picked a bad example.

Reply:
@SimenB `@types/node` should be a dev dependency, and maybe also an optional peer dependency, although I usually don't do that as it's expected that the user has that anyway if they use TypeScript.

Reply:
I really wish TS had diagnostics to warn when `@types/x` is in the wrong place or missing based on what types are used in the final package.

Reply:
If not a dep it should be an optional peer, yeah. Might make sense for react as well in this specific case?

Reply:
I've looked briefly at [`api-extractor`](https://api-extractor.com/), I _believe_ it can warn

Reply:
Rule of thumb for when something is dev/normal dep. If you are referencing a type from `@types` as an argument, return, etc type, then it must be a normal dep (since it shows up in `.d.ts`), otherwise it's a dev dep. 

For example, this is would make `@types/debug` a normal dep:

```ts
import debug from 'debug';

class Example {
  debugger: debug.Debugger = debug('example');
}
```

Reply:
@milesj Thanks for clarifying, that's a lot clearer! So it doesn't matter if I'm exporting types from 3rd-party packages in Ink's API? It should be a normal dependency, as long as it's referenced in Ink's source, even if only in "internal" code?

Reply:
If you're re-exporting types that you expect a consumer to use, then it should be a normal dep. If there any specific examples in the source code, feel free to link them.

Reply:
So if I'm not re-exporting those types, should they still be normal dependencies?

Reply:
Only if you explicitly use a type from the package and use it in your public API (argument type, return type, property type, etc). When in doubt, you can look at the compiled `.d.ts` files.

But yeah, it would be nice if there was an eslint plugin for this.

Reply:
Ok, In that case I think this PR is good to go as-is? None of the types from dependencies I've moved to `devDependencies` are exported from Ink.

Reply:
See https://github.com/vadimdemedes/ink/pull/322#discussion_r449919189, you need at least node and react - possibly optional peer deps tho

Reply:
> @SimenB @types/node should be a dev dependency, and maybe also an optional peer dependency, although I usually don't do that as it's expected that the user has that anyway if they use TypeScript.

I think comment by @sindresorhus makes sense to me, if developer is using TypeScript, they most likely already have (and should) `@types/node` and `@types/react` installed.

Reply:
@milesj seems to share the same opinion on this too, sounds logical to me.

Reply:
Yeah, that's how I do it for most of my stuff. But if you want to add them, just do `*` versions to not conflict with consumer versions.